java/semantickernel-api/src/main/java/com/microsoft/semantickernel/hooks/KernelHooks.java

```diff
@@ -102,6 +103,17 @@ public String addPreChatCompletionHook(
         return addHook((PreChatCompletionHook) function::apply);
     }
 
+    /**
+     * Add a {@link PreToolCallHook} to the collection of hooks.
+     *
+     * @param function the function to add
+     * @return the key of the hook in the collection
+     */
+    public String addPreToolCallHook(
+        Function<PreToolCallEvent, PreToolCallEvent> function) {
+        return addHook((PreToolCallHook) function::apply);
+    }
+
```

Comment:
No PostToolCallHook? Maybe I have a function that computes degrees Celsius only and I want to convert to Farenheit.

Reply:
At the moment I did not have a usecase for PostToolCall hook, @milderhc is looking at the filtering requirement, I would leave it to him to see if it is needed for the filtering work

Reply:
Odd to have pre/post for other events and not this one.